### PR TITLE
applications: nrf5340_audio: Fix BIS release compile warnings and minor typo

### PR DESCRIPTION
**Pull request by rick1082**
Fixed release build warning for BIS gateway.
Minor log/doc fixes.

Signed-off-by: Jui-Chou Chung <jui-chou.chung@nordicsemi.no>

### CODE REVIEW

**Review by koffes — APPROVED** [review]

**Review by gWacey — APPROVED** [review]

**Comment by nordicjm** [comment]
@rick1082 this has a doc-required tag and no members of doc-owners assigned to review

File: applications/nrf5340_audio/README.rst [hunk 1]
```diff
@@ -983,7 +983,7 @@ See the following examples of the parameter usage with the command run from the
 
   .. code-block:: console
 
-     python buildprog.py -c app -b debug -d both -m internal -M
+     python buildprog.py -c both -b debug -d both -m internal -M
```
[the remark this answers is not in the document]
Comment (pirun) [reply applications/nrf5340_audio/README.rst:986]: The thing is that DFU is efficient when both cores enabled.
The line 986 is actually not working.
I proposed my idea.

- For not gating this issue fix PR, leave this line as it is.
- Create another PR to remove paragraph example 2 after milestone release.

[the remark this answers is not in the document]
Comment (greg-fer) [reply applications/nrf5340_audio/README.rst:986]: If I understand your options correctly, both would mean we'll have incorrect documentation after merging this PR. So I'd prefer to fix this with a commit, let me talk to @rick1082 and @erikrobstad

[the remark this answers is not in the document]
Comment (greg-fer) [reply applications/nrf5340_audio/README.rst:987]: Pushed a commit with my proposed rephrasing.

**Review by greg-fer — APPROVED** [review]
Please squash my commit if you find it acceptable.

**Pull request merged**